authorize/authorize.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+# For testing/staging, change urls to stagingapi.artsy.net instead of api.artsy.net
+
+# Option 1: Generate access token with redirect url
+
+# 1. First direct your users to https://api.artsy.net/oauth2/authorize?client_id=${client_id}&redirect_uri=${your_website}&response_type=code
+# 2. Extract the code from the query string after the redirect
+# 3. 
+curl -X POST https://api.artsy.net/oauth2/access_token \
+     -H 'Content-Type: application/json' \
+     -d "{ \"code\": \"${code}\", \"client_id\": \"${client_id}\", \"client_secret\": \"${client_secret}\", \"grant_type\": \"authorization_code\" }"
+
+# Option 2: Generate access token with user
+
+curl -X POST https://api.artsy.net/oauth2/access_token \
+     -H 'Content-Type: application/json' \
+     -d "{ \"client_id\": \"${client_id}\", \"client_secret\": \"${client_secret}\", \"grant_type\": \"credentials\", \"email\": \"${partner_artsy_email}\", \"password\": \"${partner_artsy_password}\", \"scope\" : \"offline_access\" }"
```

Comment:
I don't think we should recommend this to partners though? Any password-handling should be the responsibility of Artsy systems, with only oauth codes used to grant 3rd parties access. Are there cases where that's insufficient?

Reply:
Yeah, I was a bit hesitant to include this, but Artsystems is a desktop app, which means normal oauth with redirect doesn't work since there's no website to redirect to. I sent them this method instead to unblock them.

Is there a better way for desktop apps to generate access tokens?

Reply:
Ideally desktop apps can also embed some sort of web flow, but I feel ya.

I have no idea how seriously these examples will be applied, so it would give me some comfort if the oauth flow was clearly the stronger recommendation, and this suggestion a last resort. Feel free to merge once that's addressed!